### ISSUE TITLE
Kestrel can't find libdl

### ISSUE THREAD

**Issue by markrendle** [aspnet/aspnet-docker#95]
I'm sure I've done something stupid, otherwise there would already be an issue for this, but I can't get a basic application running using this image as a base.

Repro, using the Basic application from `yo aspnet` with no changes:

**Dockerfile:**

```dockerfile
FROM microsoft/aspnet

COPY project.json /app/
WORKDIR /app
RUN ["dnu", "restore"]
COPY . /app


EXPOSE 5000
ENTRYPOINT ["dnx", "-p", "project.json", "kestrel"]
```

**Output:**
```
System.DllNotFoundException: libdl
  at (wrapper managed-to-native) Microsoft.AspNet.Server.Kestrel.Networking.PlatformApis/LinuxApis:dlopen (string,int)
  at Microsoft.AspNet.Server.Kestrel.Networking.PlatformApis+LinuxApis.LoadLibrary (System.String dllToLoad) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.Server.Kestrel.Networking.Libuv.Load (System.String dllToLoad) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.Server.Kestrel.KestrelEngine..ctor (ILibraryManager libraryManager, IApplicationShutdown appShutdownService) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.Server.Kestrel.ServerFactory.Start (IServerInformation serverInformation, System.Func`2 application) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.Hosting.Internal.HostingEngine.Start () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.Hosting.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute (System.Reflection.Assembly assembly, System.String[] args, IServiceProvider serviceProvider) [0x00000] in <filename unknown>:0 
  at Microsoft.Dnx.ApplicationHost.Program.ExecuteMain (Microsoft.Dnx.Runtime.DefaultHost host, System.String applicationName, System.String[] args) [0x00000] in <filename unknown>:0 
  at Microsoft.Dnx.ApplicationHost.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute (System.Reflection.Assembly assembly, System.String[] args, IServiceProvider serviceProvider) [0x00000] in <filename unknown>:0 
  at Microsoft.Dnx.Host.Bootstrapper.RunAsync (System.Collections.Generic.List`1 args, IRuntimeEnvironment env, System.Runtime.Versioning.FrameworkName targetFramework) [0x00000] in <filename unknown>:0 
```

**Comment by davidfowl** [comment]
What os?

**Comment by muratg** [comment]
@davidfowl from the trace, it looks like linux.

**Comment by yousifh** [comment]
I'm getting the same exception message. I have tried this on an Ubuntu 14.04 and RHEL7.1 VM boxes and it's failing on both. I believe a week ago I was able to run the Docker containers successfully.

**Comment by barhun** [comment]
Try and install the package libc6-dev after updating the repos. It helps me avoid this issue to add the following line to .Dockerfile, anywhere afore the statement ENTRYPOINT.

RUN apt-get update && apt-get install -y libc6-dev

**Comment by muratg** [comment]
@ahmetalpbalkan @glennc

**Comment by ahmetalpbalkan** [comment]
I wonder what  caused CI (sample HelloMvc app) to pass and yo-generated app to fail. In any case if this consistently reproes we should just add libc to the Dockerfile.

**Comment by ahmetalpbalkan** [comment]
@barhun @markrendle do you mind sending a PR adding this `libc6-dev` dependency to the Dockerfile?

**Comment by glennc** [comment]
@anurse, is this what you were hitting with beta8?

**Comment by vassilvk** [comment]
+1
I observed the same issue on my sample app.
Installing libc6-dev fixes the issue.

**Comment by barhun** [comment]
Here is all the story.

1) The latest debian:wheezy image has seemingly removed the shared object file libdl. See: tianon/docker-brew-debian@fbf1d76bdcba758d49f5fbf5c591e5149658e991
2) @glennc has invalidated the docker build cache by adding the command for removing apt lists. See: aspnet/aspnet-docker@e0759b0f0928328814799018d7cf70586cc3bf35
3) Obviously by mistake, @ahmetalpbalkan has broken the CI test steps by commenting out the lines that should have checked the sanity of the built images. See: aspnet/aspnet-docker@554eeaa685d63452e92b83bf288a6a57d450e023

Good luck, Sherlock.

**Comment by ahmetalpbalkan** [comment]
Moving on at #98.

**Comment by ahmetalpbalkan** [comment]
Fixed via #98.
